RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -3557,17 +3561,21 @@ boolean buyUpTo(int num, item it, int maxprice)
 		return false;
 	}
 
-	int orig = num;
-	num = num - item_amount(it);
-	if(num > 0)
+	int missing = num - item_amount(it);
```

Comment:
instead of copying num to orig (aka original) and then modifying num to mean "missing". create a variable named missing and use num as original